api/src/main/java/com/datastrato/gravitino/authorization/User.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Auditable;
+import java.util.List;
+
+/** The interface of a user. The user is the entity which executes every operation. */
```

Comment:
Can you please add more description about the `User` and each fields, what's the usage and limitation?

Reply:
Done.